mc-sema/cfgToLLVM/x86Instrs_Branches.cpp

```diff
@@ -1021,57 +842,23 @@ static InstTransResult translate_CALL64r(NativeModulePtr natM,
   TASSERT(tgtOp.isReg(), "");
 
   //read the register
-  Value *fromReg = R_READ<64>(block, tgtOp.getReg());
+  Value *fromReg = R_READ<width>(block, tgtOp.getReg());
 
   Module *M = block->getParent()->getParent();
   const std::string &triple = M->getTargetTriple();
   if (triple != WINDOWS_TRIPLE) {
-    x86_64::writeFakeReturnAddr(block);
+      if(width == 64) {
```

Comment:
There is no real return address. Mcsema doens't transfer control flow based on the return addr in the translated stack, so the slot goes unused, but is needed to make the stack align as the function expects. The fake one is there since it made catching off-by-4 stack errors a lot easier.